### PR TITLE
Make update-mirror.yml trigger builds on new tags

### PR DESCRIPTION
**Pull request by jwodder**
For #8.

### CODE REVIEW

File: .github/workflows/update-mirror.yml [hunk 1]
```diff
@@ -14,11 +14,33 @@ jobs:
...
+      - name: Fetch updates from official repository
         run: |
           git config pull.ff only
           git remote add official git://git.kitenet.net/git-annex
           git pull --tags official master
-          git push --tags origin upstream/master
+
+      - name: Determine new tags
+        run: |
+          comm -23 <(git tag | sort) \
```
Comment (yarikoptic) [inline .github/workflows/update-mirror.yml:25]: cool -- didn't know about `comm`, nice

File: .github/workflows/update-mirror.yml [hunk 2]
```diff
@@ -14,11 +14,33 @@ jobs:
...
+                   <(git ls-remote --tags --refs origin \
+                    | awk '{print $2}' \
+                    | xargs git for-each-ref --format '%(refname:strip=2)' \
+                    | sort) > new-tags.txt
+
+      - name: Push new objects to mirror
+        run: git push --tags origin upstream/master
```
Comment (yarikoptic) [inline .github/workflows/update-mirror.yml:32]: let's just push `upstream/master` here but push tags only after uploading the built artifact.  This way if build (but not testing) fails, we have a chance to adjust our workflow and cause rebuild of all the "yet to be done" tags.  WDYT?

**Review by yarikoptic — CHANGES_REQUESTED** [review]
minor suggestion (could be argued against ;))

File: .github/workflows/update-mirror.yml (hunk 2 above)
Comment (jwodder) [reply .github/workflows/update-mirror.yml:32]: @yarikoptic So are you expecting the build workflows to only ever be manually/programmatically dispatched using tags, not commit hashes or other refspecs?  The build workflows will need to be edited to pull the input tags from the official repository and then push them at the end, something that doesn't apply to general commitishes.

Comment (yarikoptic) [reply .github/workflows/update-mirror.yml:32]: Ah, right, I see... oh well, then let's keep as is to keep it simple(r) for now

**Pull request merged**